restangular/restangular.d.ts

```diff
@@ -90,11 +90,20 @@ declare namespace restangular {
     withConfig(configurer: (RestangularProvider: IProvider) => any): IService;
     restangularizeElement(parent: any, element: any, route: string, collection?: any, reqParams?: any): IElement;
     restangularizeCollection(parent: any, element: any, route: string): ICollection;
-    service(route: string, parent?: any): IService;
+    service(route: string, parent?: any): IScopedService;
     stripRestangular(element: any): any;
     extendModel(route: string, extender: (model: IElement) => any): void;
     extendCollection(route: string, extender: (collection: ICollection) => any): void;
   }
+  
+  interface IScopedService extends IService {
+    one(id: number): IElement;
+    one(id: string): IElement;
+    post(elementToPost: any, queryParams?: any, headers?: any): IPromise<any>;
```

Comment:
The problem is that if someone writes `myScopedService.post(3)` then `IPromise<number>` will not be inferred because the `any` overload comes first. Thus `post<number>` could only be used explicitly. If the promise doesn't actually return the same type as `elementToPost`, then the *second* overload shouldn't exist.

On another note I noticed that `var firstAccount` in the tests is of type `any` and so the typings aren't really getting tested. But that's a job for another PR.